torch/onnx/utils.py

```diff
@@ -206,6 +206,8 @@ def _optimize_graph(graph, operator_export_type, _disable_torch_constant_prop=Fa
         torch._C._jit_pass_onnx_scalar_type_analysis(graph)
         torch._C._jit_pass_lint(graph)
 
+        dynamic_fold = (dynamic_axes != {})
+        torch._C._jit_pass_onnx_fold_if(graph, dynamic_fold)
```

Comment:
I think you can determine if dynamic_fold exist by checking shape in type. It is dynamic if there are `Symbol`.